crates/tabby-scheduler/src/index.rs

```diff
@@ -1,21 +1,23 @@
 use std::{fs, io::IsTerminal, path::Path};
 
+use ignore::Walk;
 use kdam::BarExt;
+use kv::Batch;
 use tabby_common::{
     config::RepositoryConfig,
     index::{register_tokenizers, CodeSearchSchema},
     path, SourceFile,
 };
-use tantivy::{directory::MmapDirectory, doc, Index};
+use tantivy::{directory::MmapDirectory, doc, Index, Term};
 use tracing::{debug, warn};
 
-use crate::{code::CodeIntelligence, utils::tqdm};
+use crate::{cache::CacheStore, code::CodeIntelligence, utils::tqdm};
 
 // Magic numbers
 static MAX_LINE_LENGTH_THRESHOLD: usize = 300;
 static AVG_LINE_LENGTH_THRESHOLD: f32 = 150f32;
 
-pub fn index_repositories(_config: &[RepositoryConfig]) {
+pub fn index_repositories(cache: &mut CacheStore, config: &[RepositoryConfig]) {
```

Comment:
split the body into two function for cleanness

1. add_changed_documents.
2. remove_staled_documents.